magicbot/magicrobot.py

```diff
@@ -433,31 +433,57 @@ def _setup_vars(self, cname, component):
         
         component_type = type(component)
         
+        # Iterate over variables with type annotations
+        for n, inject_type in getattr(component, '__annotations__', {}).items():
+            attr = getattr(component_type, n, True)
+
+            # If the variable is private or has a value associated
+            # hasattr will return True only if the variable has been initialized
+            if n.startswith('_') or hasattr(component_type, n):
```

Comment:
I think you want to handle the case where the variable name starts with `_` separately.